### PR TITLE
US-308683 - 8.2+ Support for docker-pega-web-ready

### PR DESCRIPTION
**Pull request by Saurabh-16**
Please review the changes for docker-pega-web-ready.
The code has been brought in sync with internal git master branch.

### CODE REVIEW

**Comment by dcasavant** [comment]
Support for issue #18

**Comment by cdancy** [comment]
LGTM. Do we know or is it documented how we are going to update this code and/or version it?

**Comment by Saurabh-16** [comment]
> LGTM. Do we know or is it documented how we are going to update this code and/or version it?

The code is open source now on github . Users are free to raise bugs, PRs . This will be reviewed and merged accordingly.
This is fried image and supposed to be version agnostic. User will mostly extend it based on their requirements.They are suppose to mount their custom prweb.war when they run the image.

**Comment by cdancy** [comment]
@dcasavant are we ripping out the code from the platform as well as part of this or is there some process in place where we will occasionally (maybe per release?) push our code back upstream to this project?

File: README.md [hunk 1]
```diff
@@ -1,73 +1,120 @@
...
+Pega Docker Image
+===========
 
-This project produces a Docker image that you can use as a base image to create a complete Docker image for running Pega 7.  
+This project is a slim version of pega-web docker image that runs pega application expecting prweb.war to be mounted while performing docker run.
 
-This image is based on Tomcat 7 which is based on OpenJDK's Java 8 image. 
 
-Supported features:
+# Build
```
Comment (taz-mon) [inline README.md]: Maybe make it Building your image

File: Dockerfile [hunk 2]
```diff
@@ -1,66 +1,134 @@
...
+ENTRYPOINT ["/scripts/docker-entrypoint.sh"]
+CMD ["run"]
 
-# Expose the list of Hazelcast ports
-EXPOSE 5701-5710
+# Expose required ports
 
-# Expose Ignite port
-EXPOSE 47100
+# HTTP is 8080, SMA is 8090, JMX is 9001, Hazelcast is 5701-5710, Ignite is 47100, REST for Kafka is 7003
```
[the remark this answers is not in the document]
Comment (Saurabh-16) [reply Dockerfile]: Ok..will do this change.

File: Dockerfile [hunk 3]
```diff
@@ -1,66 +1,134 @@
...
+# to the "constructed" JDBC URL
+ENV JDBC_URL='' \
+    DB_USERNAME='' \
+    DB_PASSWORD='' \
+    JDBC_CLASS=''
+
+# Load a default PostgreSQL driver on startup
+ENV JDBC_DRIVER_URI=''
+
+#ADD https://jdbc.postgresql.org/download/postgresql-42.1.1.jre7.jar ${CATALINA_HOME}/lib/
```
[the remark this answers is not in the document]
Comment (Saurabh-16) [reply Dockerfile]: I will remove this line from DockerFile as this is already mentioned in the README.md

File: README.md [hunk 4]
```diff
@@ -1,73 +1,114 @@
...
 
-Although this code supports running Pega7 in a tomcat based docker container, it does not mean that Pega7 can be used in a docker container clustering environment as is, such as Docker Swarm and Kubernetes.
 
-# Using this image
+# Run
 
-The image itself is not runnable directly because it does not come with the Pega 7
- web applications.  Therefore you must use this image as a base to construct an 
- executable Docker image.
+**Mounting Options of prweb**
```
[the remark this answers is not in the document]
Comment (Saurabh-16) [reply README.md]: Yeah.Will be updating README

File: tomcat-bin/setenv.sh [hunk 5]
```diff
@@ -22,48 +28,24 @@ CATALINA_OPTS=""
...
 CATALINA_OPTS="${CATALINA_OPTS} -DmaxThreads=${MAX_THREADS}"
 
-# Database connection settings
-CATALINA_OPTS="${CATALINA_OPTS} -DdbName=${DB_NAME}"
-CATALINA_OPTS="${CATALINA_OPTS} -DdbHostname=${DB_HOST}"
-CATALINA_OPTS="${CATALINA_OPTS} -DdbPort=${DB_PORT}"
-CATALINA_OPTS="${CATALINA_OPTS} -DdbUsername=${DB_USERNAME}"
-CATALINA_OPTS="${CATALINA_OPTS} -DdbPassword=${DB_PASSWORD}"
-
-# JDBC connection settings
```
[the remark this answers is not in the document]
Comment (Saurabh-16) [reply tomcat-bin/setenv.sh]: I just brought them in sync with platform code. If required , I can add them back.

File: README.md [hunk 6]
```diff
@@ -1,73 +1,114 @@
...
 
-Supported features:
+# Build
 
-* PR Web application container configuration (port 8080)
-* PR System Management application container configuration (port 8090)
-* Remote JMX management support (port 9001)
+Docker (you may replace pega-tomcat with the name you wish to give the resulting image):
 
-# Usability note
```
[the remark this answers is not in the document]
Comment (Saurabh-16) [reply README.md]: ok.

File: Dockerfile [hunk 7]
```diff
@@ -1,66 +1,134 @@
-# Base image on tomcat 7 with OpenJDK JRE 8
-FROM tomcat:7-jre8
+# Dockerfile
+# Copyright (c) 2017  Pegasystems Inc.
```
[the remark this answers is not in the document]
Comment (Saurabh-16) [reply Dockerfile]: ok. I will remove this line

File: Dockerfile [hunk 8]
```diff
@@ -1,66 +1,134 @@
...
 
-# Setup global database variables
-ENV DB_USERNAME=pega \
-    DB_PASSWORD=pegasys \
-    DB_HOST=postgresql \
-    DB_PORT=5432 \
-    DB_NAME=pega
+LABEL vendor="Pegasystems Inc." \
+      name="Pega Tomcat Node" \
+      version="1.0.0"
```
[the remark this answers is not in the document]
Comment (Saurabh-16) [reply Dockerfile]: ok..Will increment it by 1.

File: Dockerfile [hunk 9]
```diff
@@ -1,66 +1,134 @@
...
+
+ENV PEGA_DOCKER_VERSION=1.0
```
[the remark this answers is not in the document]
Comment (Saurabh-16) [reply Dockerfile]: Will make it 2.0

**Comment by dcasavant** [comment]
@cdancy The existing internal source will be refactored to use this repo.

**Comment by cdancy** [comment]
@dcasavant interesting. Is the thinking then we are going to checkout/clone this project for every build (which there are many done throughout the day)? Just trying to wrap my head around how this is going to work and the efficiency behind it.

**Pull request merged**